### ISSUE TITLE
Questions about training data

### ISSUE THREAD

**Issue by Sanster** [eragonruan/text-detection-ctpn#215]
Thanks for your code and dataset!

I have two questions about training data:
1. Some images' short side length is not equal to 600(e.g. img_1651), but the label's width is `16`, so when [get_minibatch ](https://github.com/eragonruan/text-detection-ctpn/blob/master/lib/roi_data_layer/minibatch.py#L31), each box's width in `gt_boxes`  will not equal to 16. Will this behavior affect the text detect performance?
2. Should we remove labels like following (red part) to get better performance?

img_1652.jpg
![img_1652](https://user-images.githubusercontent.com/3998421/43706196-4ab2fbc8-9997-11e8-820a-a0bd70232a21.png)

img_1676.jpg
![123](https://user-images.githubusercontent.com/3998421/43706792-e56693fe-9998-11e8-8f6e-c91399fe4282.png)

**Comment by Sanster** [comment]
There are also some images have wrong labels, like img_3591.jpg

![image](https://user-images.githubusercontent.com/3998421/43749695-b52e8450-9a27-11e8-9f5c-8760d5c9eef0.png)

**Comment by eragonruan** [comment]
@Sanster good job!

**Comment by Sanster** [comment]
@interxuxing Maybe
- **No side-refinement part**
- **Different way from Conv5 to BLSTM see https://github.com/eragonruan/text-detection-ctpn/issues/193**
- The training data is different
- Use `adam`. Origin CTPN use SGD
- ...

**Comment by interxuxing** [comment]
@Sanster Thank you for your prompt reply. I figured out that there are several differences between this implementation and the original paper.

As you have explored, the MTL datasets is not very clean/accurate for training. Do you think using the synthesized data such as [SynthText](https://github.com/ankush-me/SynthText) is useful for better performance, though the text are synthesized and embedded in some template image.

**Comment by Sanster** [comment]
@interxuxing I think it worth a try

**Comment by Sanster** [comment]
- Cleaned data MLT17(latin+chinese) + ICDAR13: [google drive](https://drive.google.com/file/d/1S9K9NKkA0RYlBswCfyUI0dv_fI4r5bcX/view)
- Code for split text line by `minAreaRect`: [mlt17_to_voc.py](https://github.com/Sanster/tf_ctpn/blob/master/tools/mlt17_to_voc.py)

**Comment by saicoco** [comment]
"origin CTPN" means the model release by paper, or caffe repo

**Comment by Sanster** [comment]
@saicoco The result of "origin CTPN" if from ICDAR13 result page

**Comment by saicoco** [comment]
ok. get it

**Comment by Wangweilai1** [comment]
@Sanster I have a problem for it, if you remove labels like following (red part) , In the process of PRN, Those small text may be selected as Negative samples, so that is there a problem for training CTPN？

**Comment by Sanster** [comment]
@Wangweilai1 I think vertical words(not suitable for CTPN), very small words(can't recognize by human) should be negative examples, or we can create a ignore mask like in [EAST](https://github.com/argman/EAST/blob/ab97939783901b7e22ff55e151964e159d1627b9/icdar.py#L485). Not sure which way is better.

**Comment by Wangweilai1** [comment]
Yea,may be create a ignore mask is a good idea, In order to avoid text be selected as negative. Thanks for you answer!

**Comment by toxic2m** [comment]
Hi @Sanster @eragonruan 

Could you please share how you guys draw the ground truth boxes on training image?
I am analyzing the difference between this model and CTPN original model.
Zhi Tian(CTPN author) suggested me to check your dataset, maybe the ground truth is too wider than text content.
Thanks.

**Comment by boris-lb** [comment]
@Sanster can you tell me how to calculate the precision and recall? thanks !

**Comment by snowwindy** [comment]
@Sanster @eragonruan  你好，我使用的自己的数据集，手写体，且文字方向不固定，检测结果效果不好，特别是竖直文字，想问一下，我如何修改算法能够适合竖直文字的检测？或者有没有更好的算法，多谢

**Comment by magicxiaobai** [comment]
![default](https://user-images.githubusercontent.com/33364909/50720868-64bfd600-10f0-11e9-9a31-a3d185e52656.png)
这个怎么改？好奇怪

**Comment by northeastsquare** [comment]
> @Sanster @eragonruan 你好，我使用的自己的数据集，手写体，且文字方向不固定，检测结果效果不好，特别是竖直文字，想问一下，我如何修改算法能够适合竖直文字的检测？或者有没有更好的算法，多谢

ctpn只是用来检测水平文字的

**Comment by northeastsquare** [comment]
> ![default](https://user-images.githubusercontent.com/33364909/50720868-64bfd600-10f0-11e9-9a31-a3d185e52656.png)
> 这个怎么改？好奇怪

数据集有问题，把对应错误的变量打印出来，检查一下

**Comment by hsonetta** [comment]
Hi , How did you annotate your own data while preparing the training dataset ?

**Comment by magicxiaobai** [comment]
> I am trying to clean the data and recreate the anchor labels from MLT17 according to the `minAreaRect` of a text line. Not sure whether the training result will be better or not, but I think it worth a try. I will release the cleaned data at [tf_ctpn](https://github.com/Sanster/tf_ctpn) once finished.
> 
> Split text line by bounding box:
> ![1036_old](https://user-images.githubusercontent.com/3998421/43936923-8aca8d04-9c8d-11e8-9e8d-3271ec1f205f.jpg)
> 
> Split text line by minAreaRect:
> ![1036](https://user-images.githubusercontent.com/3998421/43936935-964fe368-9c8d-11e8-8d93-39f0d486fbc2.jpg)
下载不了谷歌网盘的文件，所以找不到清理之后的数据，请问，可以提供吗？

**Comment by chiyandetaotie** [comment]
> @Sanster can you tell me how to calculate the precision and recall? thanks !

Have you solve the problem? I have the same questions, could you share the methods?
